### PR TITLE
Properly encapsulate providers to allow for different detonation UUIDs

### PR DESCRIPTION
**Pull request by christophetd**
## Current state

"Providers" (i.e. instances of the AWS SDK, GCP SDK etc.) are currently globally shared. Each attack technique retrieves the provider it needs by using the global `providers` namespace:

```go
// https://github.com/DataDog/stratus-red-team/blob/main/v2/internal/attacktechniques/aws/credential-access/ssm-retrieve-securestring-parameters/main.go#L55
func detonate(params map[string]string) error {
	ssmClient := ssm.NewFromConfig(providers.AWS().GetConnection())
        // ...
}
```

When Stratus Red Team detonates an attack technique, it injects an UUID in the user-agent of the relevant SDK. This UUID is also globally accessible through `providers.UniqueExecutionId`.

## The problem

The detonation UUID is currently global **per process**. For instance, if we detonate multiple Stratus Red Team attack techniques through the programmatic interface, **all the detonations have same UUID**.

This is a problem because it defeats the very purpose of this UUID mechanism: creating a causation between a detonation and the resulting logs.

## The solution

* Remove all global variables related to providers
* Introduce a new interface `CloudProviders` and structure `CloudProvidersImpl` holding references to the appropriate providers:

```go
// CloudProviders provides a unified interface to access the various cloud providers SDKs
type CloudProviders interface {
	AWS() *providers.AWSProvider
	K8s() *providers.K8sProvider
	Azure() *providers.AzureProvider
	GCP() *providers.GCPProvider
}
```

* Instantiate providers as requested, to avoid e.g. throwing AWS authentication errors if we detonate an Azure attack technique
* Modify the signature of the `detonate` and `revert` attack technique functions to inject a `CloudProviders` (dependency injection):

```go
func detonate(params map[string]string, providers stratus.CloudProviders) error {
	ssmClient := ssm.NewFromConfig(providers.AWS().GetConnection())
        // ...
}
```

## Retro-compatibility

This should be fully retro-compatible. 

## Manual tests

- [x] GCP: UUID is properly injected
- [x] AWS: UUID is properly injected
- [ ] Azure: UUID is properly injected
- [x] K8s: UUID is properly injected
- [ ] If authenticated only against GCP, no errors are thrown when detonating an AWS attack technique
- [ ] If authenticated only against AWS, no errors are thrown when detonating an AWS attack technique
- [ ] If authenticated only against Azure, no errors are thrown when detonating an AWS attack technique
- [ ] If authenticated only against K8s, no errors are thrown when detonating an AWS attack technique

### CODE REVIEW

File: v2/pkg/stratus/runner/runner.go [hunk 1]
```diff
@@ -203,9 +211,9 @@ func (m *Runner) setState(state stratus.AttackTechniqueState) {
 	m.TechniqueState = state
 }
 
-// GetUniqueExecutionId returns an unique execution ID, unique per run of Stratus Red Team (not for each TTP detonated)
+// GetUniqueExecutionId returns an unique execution ID, unique for each runner instance
 func (m *Runner) GetUniqueExecutionId() string {
-	return providers.UniqueExecutionId.String()
```
Comment (christophetd) [inline v2/pkg/stratus/runner/runner.go:208]: note: This used to be the global unique execution ID that's now properly encapsulated

**Review by juliendoutre — APPROVED** [review]
The changes make sense to me ✅ 

I noticed you call the `VerifyPlatformRequirements` function before most commands are actually executed (warmup, detonate and revert). This function currently initialises its own providers with a different UUID than the ones used later during the command execution (handled through the runner's logic). Is it on purpose? Would it make sense to use the same UUID for both? Or maybe it does not matter?

**Comment by christophetd** [comment]
Will add a comment - in that case it doesn't matter as it's only for checking authentication so we don't care about the UUID

**Comment by christophetd** [comment]
Also reviewed by @rcobb-scwx who says the PR looks fine so I'll go ahead, merge it and release shortly

**Pull request merged**